### PR TITLE
Add snippet for impacts on parquet file of data type and compression algorithm

### PR DESCRIPTION
**Pull request by ThAccart**
Not 100% sure this leads used to the point we want to demonstrate.
Could you have a look at it ?

### CODE REVIEW

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Databricks notebook source
+// Spark: 3.5.1
+// Local: --driver-memory 1G --master 'local[2]' --conf spark.ui.retainedJobs=2
+
+// COMMAND ----------
+
+/*
+This snippet shows who data type for numerical information and compression can affect Spark.
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Typo: `how`.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+# Symptom
+Storage needs does not match with expectations, for example is higher in output after filtering than in input.
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Typo, `needs do not match...`.

Comment (mauriciojost) [reply spark3/parquet-compression.choose-types-wisely.scala]: I'd suggest `for example volume is higher`.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+# Explanation
+There is difference in in type when reading the data and type when writing it, causing a loss of compression perfomance.
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Typo, double `in`.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+*/
+
+// COMMAND ----------
+
+// We are going to demonstrate our purpose by converting the same numerica data into different types,
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Typo, `numerical`.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+// and write it in parquet using different compression
+
+
+// Here are the type we want to compare
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Typo: `types`.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  for (parquetCompressionName <- allParquetCompressions ){
+    val fileName=s"1M/Parquet_${numericTypeName}_${parquetCompressionName}"
+    print (".")
+    spark.sparkContext.setJobGroup("Write data",s"Write ${numericTypeName} with compression ${parquetCompressionName}")
+    rndDF.selectExpr(s"cast(value as $numericTypeName )")
+      .write
+      .option("compression", parquetCompressionName)
+      .format("parquet")
+      .mode("overwrite")
+      .save(fileName)
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: I'd reuse the mechanism for creating random and disposable temporary directories that is already present in the other notebooks. This way temporary files land in the predefined and already set up trash directory. See the other snippets for reference (they use an uuid, you can use the same imports and snippet).

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+//search for sum of files size in kiloBytes, per parent folder:
+val sizeOnDisk= hadoopFileSystem.globStatus(new org.apache.hadoop.fs.Path("./1M/**/part*"))
+  .filter(o=>o.isFile)
+  .map(o=> (o.getLen/1024, o.getPath.getParent.toString))
+  .groupBy(_._2)
+  .mapValues(_.map(_._1).sum).toSeq.sortBy(_._2)
+
+println("part* files sizes (in kB):")
+sizeOnDisk.foreach( o=>println ( s"${o._1}\t${o._2}\tkB"))
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: Using implicits, you can use `toDF` to display a more friendly table:
image

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+// COMMAND ----------
+
+// now we can also add a check on the effect of choosing a specific number type on the obtained values
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala]: I'd rephrase this as follows:
```
// Now let's see how long it takes to read and process such data when using different compression and numerical data formats .
```

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+
+println("Time to read and calculating aggregated value")
+readInformations.foreach(readInformation=> println ( s"${readInformation._1}\t${readInformation._2} ms\t(agg value was ${readInformation._3})" ))
+
+
+
+// COMMAND ----------
+
+// Now you should also check how the variety of values affects compression :
```
Comment (mauriciojost) [inline spark3/parquet-compression.choose-types-wisely.scala:176]: I'd encourage you to add some conclusions here so that people reading the snippet can take action and understand what to expect with their change.

File: spark3/parquet-compression.choose-types-wisely.scala [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+/*
+This snippet shows how data type for numerical information and compression can affect Spark.
+
+# Symptom
+Storage needs do not match with expectations, for example volume is higher.
+
+# Explanation
+There is difference in type when reading the data and type when writing it, causing a loss of compression perfomance.
+
```
Comment (generoso) [inline spark3/parquet-compression.choose-types-wisely.scala:15]: Could you please add the `# What to aim for concretely` section, as in other snippets? E.g. choose the right data type according to your needs and taking performance into account. You could reference the example you gave below.

**Review by generoso — APPROVED** [review]

**Comment by mauriciojost** [comment]
Thanks for the changes!

**Review by mauriciojost — APPROVED** [review]
Thanks for the changes!
